splunk-otel-android/src/main/java/com/splunk/rum/ConnectionUtil.java

```diff
@@ -102,26 +103,19 @@ CurrentNetwork getActiveNetwork() {
         return currentNetwork;
     }
 
-    void setInternetStateListener(ConnectionStateListener listener) {
-        connectionStateListener = listener;
+    void addNetworkChangeListener(NetworkChangeListener listener) {
+        listeners.add(listener);
     }
 
     private class ConnectionMonitor extends ConnectivityManager.NetworkCallback {
 
         @Override
         public void onAvailable(@NonNull Network network) {
-            Log.d(SplunkRum.LOG_TAG, "onAvailable: ");
             CurrentNetwork activeNetwork = refreshNetworkStatus();
-            ConnectionStateListener connectionStateListener =
-                    ConnectionUtil.this.connectionStateListener;
-            if (connectionStateListener != null) {
-                connectionStateListener.onAvailable(true, activeNetwork);
-                Log.d(
-                        SplunkRum.LOG_TAG,
-                        "  onAvailable: isConnected:"
-                                + isOnline()
```

Comment:
With this removal, is `isOnline()` still used? I couldn't really find usages of it anymore but might have just missed it. Probably just delete it.

Reply:
Nope, I missed that one -- will remove 👍